java/ql/src/semmle/code/java/frameworks/ApacheHttp.qll

```diff
@@ -13,3 +13,32 @@ class ApacheHttpEntityGetContent extends Method {
     this.getName() = "getContent"
   }
 }
+
+/**
+ *  A class derived from the `HttpRequestBase` or the `BasicHttpRequest`
+ *  class of the Apache Http Client `org.apache.http` library
+ */
+class TypeApacheHttpRequestBase extends RefType {
+  TypeApacheHttpRequestBase() {
+    this
+        .getASourceSupertype*()
+        .hasQualifiedName("org.apache.http.client.methods", "HttpRequestBase") or
+    this.getASourceSupertype*().hasQualifiedName("org.apache.http.message", "BasicHttpRequest")
+  }
+}
+
+/*
+ * Any class which can be used to make an HTTP request using the Apache Http Client library
+ * Examples include `HttpGet`,`HttpPost` etc.
+ */
+
+class TypeApacheHttpRequest extends Class {
```

Comment:
This class is almost equal to `TypeApacheHttpRequestBase` - they both include subclasses of `HttpRequestBase` and `BasicHttpRequest`, but this class excludes just those two superclasses. Is there an important reason for this distinction? Otherwise let's just use one class. (And the simpler name `ApacheHttpRequest` seems fine.)

Reply:
I have directly inlined the class now.